### DISCUSSION TITLE
Embedding code or hard code it in C API?

### DISCUSSION THREAD

**Discussion by stefanos82** [edubart/nelua-lang#211] [Q&A] [answered]
I know you hexdump `luainit.lua` in `luainit.h` that gets called from `luainit.c` in `src`.

I was wondering...

Let's assume that I hard code it in C API to make the actual calls.

Would there be any benefit to such action or the performance is more or less the same?

What's your personal experience on this matter?

**Comment by edubart** [comment, marked as answer]
The performance of the compiler would be a little better on startup if `luainit.lua` logic was entirely implemented in C. But it's so negligible, I guess would improve some microseconds of `nelua-lua` initialization. However technically would be a cleaner solution, but it's not something worth doing, there are way more interesting things to improve.
